jax/_src/lax/lax.py

```diff
@@ -4534,23 +4534,35 @@ def _dilate_shape(shape, dilation):
 def _ceil_divide(x1, x2):
   return -np.floor_divide(np.negative(x1), x2)
 
+
+jax_enum_names = [*[es.name for es in xla_client.PaddingType], 'PYTORCH_SAME']
+PaddingType = enum.Enum(
+    'JaxPaddingType',
+    jax_enum_names,
+    start=0,
+)
+
 def padtype_to_pads(in_shape, window_shape, window_strides, padding):
   """Convert padding string to list of pairs of pad values."""
-  PaddingType = xla_client.PaddingType
-
   if isinstance(padding, str):
-    mapping = {'VALID': PaddingType.VALID, 'SAME': PaddingType.SAME}
     try:
-      padding = mapping[padding.upper()]
+      padding = PaddingType[padding.upper()]
     except KeyError as err:
       msg = "Unrecognized padding type: expected 'VALID' or 'SAME', got {}."
       raise RuntimeError(msg.format(padding)) from err
 
-  if padding == PaddingType.SAME:
+  if padding in [PaddingType.SAME, PaddingType.PYTORCH_SAME]:
     out_shape = _ceil_divide(in_shape, window_strides)
-    pad_sizes = np.maximum(0, (out_shape - 1) * window_strides +
-                                window_shape - in_shape)
-    return [(pad_size // 2, pad_size - pad_size // 2) for pad_size in pad_sizes]
+    pad_sizes = np.maximum(
+        0,
+        (out_shape - 1) * window_strides + window_shape - in_shape,
+    )
+    if padding == PaddingType.SAME:
+      return [(pad_size // 2, pad_size - pad_size // 2)
+              for pad_size in pad_sizes]
+    else:
+      return [((pad_size + 1) // 2, pad_size - (pad_size + 1) // 2)
+              for pad_size in pad_sizes]
```

Comment:
Those two return statements only differ in `pad_size` vs `pad_size + 1`, so you could deduplicate them even more.

Reply:
I relied on two sources.
1. [PyTorch implementation](https://github.com/pytorch/pytorch/blob/master/torch/nn/modules/conv.py#L120)
2. [haiku.pad.same](https://github.com/deepmind/dm-haiku/blob/main/haiku/_src/pad.py#L37) implementation which is equivalent to the first one.
3. [Converting tensorflow model to pytorch: issue with padding](https://discuss.pytorch.org/t/converting-tensorflow-model-to-pytorch-issue-with-padding/84224) blogpost